lnd_methods/offchain/estimate_route_fee.js

```diff
@@ -0,0 +1,85 @@
+const asyncAuto = require('async/auto');
+const {returnResult} = require('asyncjs-util');
+
+/** Estimate routing fees based on an invoice.
+
+  Requires `offchain:read` permission
+
+  {
+    lnd: <Authenticated LND API Object>
+    request: <BOLT 11 Payment Request String>
+    deadline: <Optional Timeout in Milliseconds Number> 
+  }
+
+  @returns via cbk or Promise
+  {
+    mtoken: <Route Fee Millitokens String>,
+    timeout: <Time Lock Block Height Delay String>
+    failure_reason: <Failure Reason String>
+  }
+*/
+module.exports = (args, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!args.lnd || !args.lnd.router) {
+          return cbk([400, 'ExpectedAuthenticatedLndForRouteFeeEstimate']);
+        }
+
+        if (!args.request) {
+          return cbk([400, 'ExpectedPaymentRequestStringForRouteFeeEstimate']);
+        }
+        
+        if (args.timeout > 86400000) {
+          return cbk([400, 'ExpectedDeadlinLessThanOneDayForRouteFeeEstimate']);
+        }
+
+        if (args.timeout < 1) {
+          return cbk([400, 'ExpectedDeadlinGreaterThanZeroForRouteFeeEstimate']);
+        }
+
+        if (!args.deadline) {
+          args.deadline = 60000;
+        }
+
+        return cbk();
+      },
+
+      // Estimate route fee and return a successful routing fee and time lock delay
+      estimateRouteFee: ['validate', ({}, cbk) => {
+        const request = args.request;
+        const timeout = args.timeout || 60000;
+        return args.lnd.router.estimateRouteFee({request, timeout},
```

Comment:
the passed timeout would be in milliseconds but the argument to grpc would be in seconds so it needs to be converted